src/test/java/api/support/fakes/FakeStorageModule.java

```diff
@@ -61,7 +62,7 @@
   private final JsonSchemaValidator recordValidator;
   private final String batchUpdatePath;
   private final Function<JsonObject, JsonObject> batchUpdatePreProcessor;
-  private final Function<JsonObject, CompletableFuture<JsonObject>> recordPreProcessor;
+  private final Collection<BiFunction<JsonObject, JsonObject, CompletableFuture<JsonObject>>> recordPreProcessors;
```

Comment:
```suggestion
  private final List<BiFunction<JsonObject, JsonObject, CompletableFuture<JsonObject>>> recordPreProcessors;
```

I'd use just `List` here, since the pre-processors should be applied in some order.

Reply:
Agree